File: lista.py
class Lista:
    class No:
        def __init__(self, valor, proximo=None):#dois atributos essenciais p um nó
            self.valor = valor 
            self.proximo = proximo

    def __init__(self):#crio os atributos principais da minha lista
        self.__cabeca = None
        self.__quantidade = 0
    
    def __len__(self):#para determinar a quantidade de elementos
        return self.__quantidade

    def __getitem__(self, posicao):#p consegyur acessar os elementos pela posição
        if posicao < 0 or posicao >= self.__quantidade:
            raise IndexError('Posição Inválida.')
        
        atual = self.__cabeca
        for i in range(posicao):
            atual = atual.proximo
        return atual.valor
    
    def inserir(self, posicao, valor):
        novo = self.No(valor) #crio o meu elemento que desejo inserir
        if posicao <= 0 or self.__cabeca is None:
            novo.proximo = self.__cabeca
            self.__cabeca = novo
        else:
            atual = self.__cabeca#percorrer começando pela cabeça
            for i in range(posicao - 1):
                if atual.proximo is None:
                    break
                atual = atual.proximo
            novo.proximo = atual.proximo
            atual.proximo = novo
        self.__quantidade += 1

    def remover(self, valor):
        atual = self.__cabeca
        anterior = None
        while atual is not None:
            if atual.valor == valor:
                if anterior is None:
                    self.__cabeca = atual.proximo
                else:
                    anterior.proximo = atual.proximo
                self.__quantidade -= 1
                
                atual.proximo = None
                atual.valor = None
                return
                
            anterior = atual
            atual = atual.proximo

    def printar(self):
        atual = self.__cabeca
        while atual is not None:
            print(atual.valor, end=" ")
            atual = atual.proximo
        print()

def processar_arquivo(file_name):
    lista = Lista()  # Cria uma instância da classe Lista
    with open(file_name, 'r') as file:  # Abre o arquivo para leitura
        linhas = file.readlines()  # Lê todas as linhas do arquivo
        
        # Processa os elementos iniciais da lista
        elementos_iniciais = map(int, linhas[0].split())  # Converte a primeira linha em uma lista de inteiros
        for valor in elementos_iniciais:
            lista.inserir(len(lista), valor)  # Insere cada valor na lista na última posição
        
        num_operacoes = int(linhas[1].strip())  # Converte a segunda linha para obter o número de operações
        
        # Processa as operações
        for i in range(2, 2 + num_operacoes):  # Para cada operação no arquivo
            comando = linhas[i].strip().split()  # Divide a linha de comando em partes (por espaço)
            if comando[0] == 'A':  # Comando para adicionar
                lista.inserir(int(comando[2]), int(comando[1]))  # Insere na posição especificada
            elif comando[0] == 'R':  # Comando para remover
                lista.remover(int(comando[1]))  # Remove o valor especificado
            elif comando[0] == 'P':  # Comando para imprimir
                lista.printar()  # Imprime a lista


processar_arquivo('arq3.txt')

File: lista.java
import java.io.*;

class Node {
    int valor;
    Node proximo;

    public Node(int valor) {
        this.valor = valor;
        this.proximo = null;
    }
}

class Lista {
    private Node inicio;

    public Lista() {
        this.inicio = null;
    }

    public void inserir(int valor, int posicao) {
        Node novoNode = new Node(valor);
        if (posicao == 0) {
            novoNode.proximo = inicio;
            inicio = novoNode;
            return;
        }

        Node atual = inicio;
        for (int i = 0; atual != null && i < posicao - 1; i++) {
            atual = atual.proximo;
        }

        if (atual == null) {
            return;
        }

        novoNode.proximo = atual.proximo;
        atual.proximo = novoNode;
    }

    public void Remover(int valor) {
        Node atual = inicio, anterior = null;

        while (atual != null) {
            if (atual.valor == valor) {
                if (anterior != null) {
                    anterior.proximo = atual.proximo;
                } else {
                    inicio = atual.proximo;
                }
                atual.proximo = null;
                return;
            }
            anterior = atual;
            atual = atual.proximo;
        }
    }

    public void exibir() {
        Node atual = inicio;
        System.out.print("Lista Atual: ");
        while (atual != null) {
            System.out.print(atual.valor + " ");
            atual = atual.proximo;
        }
        System.out.println();
    }
}

public class lista {
    public static void main(String[] args) {
        Lista lista = new Lista();

        try (BufferedReader leitor = new BufferedReader(new FileReader("arq3.txt"))) {
            // Leitura dos valores iniciais
            String[] valoresIniciais = leitor.readLine().split(" ");
            for (int i = 0; i < valoresIniciais.length; i++) {
                lista.inserir(Integer.parseInt(valoresIniciais[i]), i);
            }

            // Leitura do número de ações
            int totalAcoes = Integer.parseInt(leitor.readLine().trim());

            // Processamento das ações
            for (int i = 0; i < totalAcoes; i++) {
                String[] acao = leitor.readLine().trim().split(" ");
                switch (acao[0]) {
                    case "A" -> // Adicionar
                        lista.inserir(Integer.parseInt(acao[1]), Integer.parseInt(acao[2]));
                    case "R" -> // Remover
                        lista.Remover(Integer.parseInt(acao[1]));
                    case "P" -> // Mostrar
                        lista.exibir();
                    default -> System.out.println("Ação inválida.");
                }
            }
        } catch (IOException e) {
            System.err.println("Erro ao acessar o arquivo: " + e.getMessage());
        }
    }
}
